myokit/_model_api.py

```diff
@@ -3256,7 +3258,7 @@ def state(self):
         Returns the current state of the model as a list of floating point
         numbers.
         """
-        return list(self._current_state)
+        return [float(eqn.rhs) for eqn in self.inits()]
```

Comment:
This is definitely the path of least resistance. But should we also provide users with a way to get the real state, expressions & all ?

Even without redefining the return type, we'll need to do a grep for all calls in Myokit to `state()`, and check that they get what they are expecting (i.e. did they want floats of the current state or does it make more sense for them to get expressions now that that's an option)

Reply:
agreed. Shall I define another function `state_values()` that returns the list of numbers, and then have `state()` return the equations themselves? Any existing calls to state() I can decide to either use `state()` or `state_values()` depending on what is appropriate

Reply:
ok, so in doing this I've come across a problem regarding `set_state`. This is used to set the current state of the model to do some calculations, e.g. `evaluate_derivatives`. Presumable the intent here is that you can set the current state to something that might not be the initial state? So do we need `self._current_state`, a list of floats, and `self._initial_state`, a list of expressions? So you are free to change the current state independently from the initial_state?

Reply:
The model is only supposed to have a single state (which I might want to rename initial state at some point #435). I think evaluate_derivatives only changes it as a bit of a hack (would have been in a try-finally block if I programmed this again!)
I think the answer here is for org_state to be the state-with-variables, while the subsequent set_state uses floats, and then at the end set_state(state-with-variables) is used again to restore the original state

Reply:
An even better solution might be to rewrite the whole thing to use `subst={Name(state_x): value_x, ...}` in the calls to eval().

Reply:
Shall I make a ticket for that and rewrite it? Then you can pull that in